### PR TITLE
Button state styling changes

### PR DESCRIPTION
**Pull request by iamhimateja**
https://app.asana.com/0/1181922654617025/1200441651055177/f

### CODE REVIEW

File: src/fields/Text.jsx [hunk 1]
```diff
@@ -21,15 +21,18 @@ function Text({
     conditions,
```
[the remark this answers is not in the document]
Comment (iamhimateja) [reply src/fields/Text.jsx:21]: ok

[the remark this answers is not in the document]
Comment (iamhimateja) [reply src/fields/Text.jsx]: Divided Text component into `Text` and `ButtonElement` components.

_`Button` component is already defined in react-bootstrap package_

File: src/Form.jsx [hunk 2]
```diff
@@ -3,7 +3,7 @@ import React, { useEffect, useRef, useState } from 'react';
 import ProgressBar from 'react-bootstrap/ProgressBar';
 import ReactForm from 'react-bootstrap/Form';
 import { SketchPicker } from 'react-color';
-import TagManager from 'react-gtm-module'
+import TagManager from 'react-gtm-module';
```
Comment (bo-dun-1) [inline src/Form.jsx:6]: thanks for catching these

File: src/Form.jsx [hunk 3]
```diff
@@ -704,6 +705,37 @@ function Form({
                     isFilled={isFilled}
                 />
             ))}
+            {activeStep.buttons.map((field, i) => {
+                const hoverStyles = {};
+                const selectedStyles = {};
+                if (field.hover_border_color)
```
Comment (bo-dun-1) [inline src/Form.jsx:711]: move this logic into the ButtonElement component. Also if you're naming this ButtonElement, you should rename Text to TextElement

File: src/fields/ButtonElement.jsx [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+}) {
+    const [showSpinner, setShowSpinner] = useState(false);
+
+    const elementKey = field.text;
+    const repeat = field.repeat || 0;
+
+    const delta = new Delta(field.text_formatted);
+
+    // TODO (jake): Make this in React
+    const nodes = delta.map((op, i) => {
```
Comment (bo-dun-1) [inline src/fields/ButtonElement.jsx:32]: you need to decompose logic that's being duplicated

File: src/fields/__test__/ButtonElement.spec.jsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        act(() => {
+            button = create(<ButtonElement {...props} />);
+        });
+        const tree = button.toJSON();
+
+        // Assert
+        expect(tree).toMatchSnapshot();
+    });
+
+    it('renders a text block with custom styles', async () => {
```
Comment (bo-dun-1) [inline src/fields/__test__/ButtonElement.spec.jsx:54]: renders a button?

**Review by bo-dun-1 — CHANGES_REQUESTED** [review]

File: src/fields/__test__/ButtonElement.spec.jsx (hunk 5 above)
Comment (iamhimateja) [reply src/fields/__test__/ButtonElement.spec.jsx]: updated

File: src/Form.jsx (hunk 3 above)
Comment (iamhimateja) [reply src/Form.jsx]: Done.

File: src/fields/ButtonElement.jsx (hunk 4 above)
Comment (iamhimateja) [reply src/fields/ButtonElement.jsx]: Done.

**Review by bo-dun-1 — CHANGES_REQUESTED** [review]
please rebase

File: src/utils/nodes.jsx [hunk 6]
```diff
@@ -0,0 +1,97 @@
+import React from 'react';
```
Comment (bo-dun-1) [inline src/utils/nodes.jsx:1]: can we move this to the fields folder and call it textNodes.jsx

**Review by bo-dun-1 — CHANGES_REQUESTED** [review]
can you include your testing plan here

File: src/fields/ButtonElement.jsx [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+                    width: `${field.button_width}${field.button_width_unit}`,
+                    maxWidth: '100%'
+                }}
+                css={{
+                    '&:active': selectedStyles,
+                    '&:hover': hoverStyles,
+                    '&:disabled': {
+                        cursor: 'default !important'
+                    },
+                    '&:hover:enabled': field.link
```
Comment (bo-dun-1) [inline src/fields/ButtonElement.jsx:94]: when would field.link not exist?

**Review by bo-dun-1 — CHANGES_REQUESTED** [review]

File: src/fields/TextElement.jsx [hunk 8]
```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import Delta from 'quill-delta';
+import generateNodes from './textNodes';
+
+const buttonAlignmentMap = {
```
Comment (bo-dun-1) [inline src/fields/TextElement.jsx:5]: this should only be defined one time between text and buttonelement

**Review by bo-dun-1 — CHANGES_REQUESTED** [review]

File: src/utils/nodes.jsx (hunk 6 above)
Comment (iamhimateja) [reply src/utils/nodes.jsx]: Done.

File: src/fields/ButtonElement.jsx (hunk 7 above)
Comment (iamhimateja) [reply src/fields/ButtonElement.jsx:124]: Updated the condition

File: src/fields/TextElement.jsx (hunk 8 above)
Comment (iamhimateja) [reply src/fields/TextElement.jsx:5]: extracted this to utils.

**Review by bo-dun-1 — APPROVED** [review]

**Pull request merged**